### PR TITLE
add option to switch, rather than pop to buffer

### PR DESCRIPTION
**Pull request by attila-lendvai**
forward ported echaozh's patch from:
https://github.com/echaozh/emacs_chrome/

### CODE REVIEW

**Comment by attila-lendvai** [comment]
unfortunately, with this enabled it's probably not possible to switch back to the browser window. still, some people may prefer this behavior. i'm also on the fence.

File: servers/edit-server.el [hunk 1]
```diff
@@ -134,6 +134,12 @@ If nil, the new frame will use the existing `default-frame-alist' values."
 		       (symbol :tag "Parameter")
 		       (sexp :tag "Value"))))
 
+(defcustom edit-server-pop-buffer t
+  "If not nil and edit-server-switch-buffer is nil, pop to the buffer;
```
Comment (stsquad) [inline servers/edit-server.el]: What is edit-server-switch-buffer?

**Comment by stsquad** [comment]
I'm a little confused as to the subtleties of this - reading the docs on switch-to-buffer it seems pop-to-buffer gets called depending on the circumstances. From the manual:

    If the selected window cannot display the specified buffer
    because it is a minibuffer window or strongly dedicated to
    another buffer, call ‘pop-to-buffer’ to select the buffer in
    another window.

**Comment by attila-lendvai** [comment]
hrm. i have cherry-picked it from someone else's fork while i was looking for a better behavior. ```edit-server-switch-buffer``` is probably gone since then, or it was in another patch that is not here.

the ideal behavior for me would be not opening a new frame, displaying the edit buffer in the emacs window, bringing it forward, and when committing it bringing the browser window forward (unfortunately this last part is not possible to the best of my knowledge).

i myself ended up not using it in my current setup, so feel free to reject and close this PR.

**Comment by stsquad** [comment]
If I set edit-server-new-frame to nil I certainly get the behaviour of a new Emacs window being created in the exiting frame and that coming into focus. By default it splits my current Emacs window. As for dropping the frame after the edit there may be something you can add to the post hook.

**Comment by attila-lendvai** [comment]
what i can't do from elisp is switching the OS windows.

**Comment by stsquad** [comment]
@attila-lendvai I suspect this is all too WM specific for edit-with-emacs to deal with. However if your WM is programmable you could certainly signal it to re-focus the browser from the hook. Anyway closing this pullreq. Thanks for your interest.

**Pull request closed without merging**